src/ghost.js

```diff
@@ -67,8 +69,26 @@ class GhostThread extends EventEmitter {
    *
    * @return    {Array<Object>}      users online
    */
-  async getPosts () {
-    return [...this._backlog].map(msg => JSON.parse(msg)).sort((p1, p2) => p1.timestamp - p2.timestamp)
+  async getPosts (num = null) {
+    const sorted = [...this._backlog]
+      .map(msg => JSON.parse(msg))
+      .sort((p1, p2) => p1.timestamp - p2.timestamp)
+
+    if (num) {
+      const posts = sorted.slice(-num)
+      posts.forEach(msg => {
+        this._backlog.add(JSON.stringify(msg))
+      })
+
+      return posts
+    } else {
+      const posts = sorted.filter(msg => msg.timestamp > (Math.floor(Date.now() / 1000) - POST_AGE_LIMIT))
+      posts.forEach(msg => {
+        this._backlog.add(JSON.stringify(msg))
```

Comment:
Still wondering about this. `this._backlog` will already contain all messages being added to it here.